rust-symcrypt/src/hash.rs

```diff
@@ -0,0 +1,186 @@
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use symcrypt_sys;
+
+pub const SHA256_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+pub struct Sha256State {
+    state: symcrypt_sys::SYMCRYPT_SHA256_STATE,
+}
+
+impl Sha256State {
+    pub fn new() -> Self {
+        let mut instance = Sha256State {
+            state: symcrypt_sys::SYMCRYPT_SHA256_STATE::default(),
+        };
+        unsafe {
+            symcrypt_sys::SymCryptSha256Init(&mut instance.state);
```

Comment:
It's generally good practice to add a `// SAFETY: ` comment on this `unsafe` block or function call that explains why this operation is safe. In this case it's probably just that unsafe is needed for FFI, but safety comments also commonly say things like "this pointer was verified to be correctly allocated and initialized in function X" or whatever makes sense.

Reply:
This advice applies to the other unsafe blocks in this PR too, but I won't comment on each one individually.

Reply:
I'm confused. All this module does is wrap FFI calls. Seems like adding SAFETY is redundant information.

Reply:
In this case it doesn't really add much extra information, but I think it's still good practice to do. Unsafe blocks without `SAFETY` comments always look a little suspicious to me while something like `// SAFETY: unsafe needed for FFI` let's me know that nothing fancy is meant to be going on inside that block.